src/client/opamCommands.ml

```diff
@@ -869,6 +869,16 @@ module Var_Option_Common = struct
        | `var -> OpamConfigCommand.vars_list gt
        | `option -> OpamConfigCommand.options_list gt);
       `Ok ()
+    | `All_var ->
+      (match apply with
+       | `value_wo_eq v ->
+         OpamGlobalState.with_ `Lock_none @@ fun gt ->
+         OpamConfigCommand.var_show gt v;
+         `Ok ()
+       | `empty -> assert false (* can't happen *)
+       | `value_eq _ ->
+         `Error (true, "variable setting needs a scope, \
```

Comment:
This restriction is only for `opam var` affectations, should we set switch scope by default ?

Reply:
After some discussions, no for variables